### PR TITLE
Clement-JS- week4

### PR DESCRIPTION
**Pull request by clems27**
week 4

### CODE REVIEW

File: week-4-practise/level-1/exercise-1.js [hunk 1]
```diff
@@ -38,7 +38,15 @@ var writers = [
...
-
 /*
 If you want an extra challenge, only `console.log()` the writers that are alive.
-*/
\ No newline at end of file
+*/
+writers.forEach(function (writer) {
+  if (writer.alive) {
+    console.log(
+      `Hi, my name is ${writer.firstName} ${writer.lastName}. I am ${
```
Comment (foaly-nr1) [inline week-4-practise/level-1/exercise-1.js]: Brilliant to see you using ES6 features 👏🏻 Definitely a good place to use string literals.

File: week-4-practise/level-2/exercise-1.js [hunk 2]
```diff
@@ -11,12 +11,15 @@ We made a start on this for you here:
 var bottle = {
 	volume: 0,
 	fill: function(){
+		this.volume += 100;
```
Comment (foaly-nr1) [inline week-4-practise/level-2/exercise-1.js]: Not quite right, can you spot the bug? What happens if the bottle has half full when you `fill()` it?

File: week-4/C-more-complex-objects/exercise-1.js [hunk 3]
```diff
@@ -10,8 +10,22 @@ var house = {
     currentOwner: {
       firstName: "Margaret",
       lastName: "Conway"
+    },
+    newAddress: function(newAddress){
+      this.address = newAddress;
+    },
+    NewPreviousOwners: function(newOwner){
```
Comment (foaly-nr1) [inline week-4/C-more-complex-objects/exercise-1.js]: I got quite confused by the argument name `newOwner`. Can you think of a better name?

File: week-4/C-more-complex-objects/exercise-1.js [hunk 4]
```diff
@@ -10,8 +10,22 @@ var house = {
     currentOwner: {
       firstName: "Margaret",
       lastName: "Conway"
+    },
+    newAddress: function(newAddress){
```
Comment (foaly-nr1) [inline week-4/C-more-complex-objects/exercise-1.js]: Method names ideally start on a verb so the name indicates the action. What could be a more meaningful name for this method?

File: week-4/C-more-complex-objects/exercise-3.js [hunk 5]
```diff
@@ -32,16 +32,22 @@ var parkAvenueHouse = {
 
 // returns the full name (first name + last name) of the owner of the house
 function getOwnerFullName(house) {
+  return house.currentOwner.firstName.concat(house.currentOwner.lastName)
```
Comment (foaly-nr1) [inline week-4/C-more-complex-objects/exercise-3.js]: What happens when you call this function with `kinningParkHouse`? Does it return the full name in the format you would expect?

File: week-4/D-methods/exercise-3.js [hunk 6]
```diff
@@ -8,11 +8,11 @@ var person = {
     name: "Alice",
     age: 25,
     currentAddress: "Glasgow",
-    changeAddress: (newAddress) {
+    changeAddress: function (newAddress) {
         currentAddress = newAddress;
```
Comment (foaly-nr1) [inline week-4/D-methods/exercise-3.js]: There's a bug here. Try to debug it by actually calling your method `changeAddress`.

File: week-4/D-methods/exercise-4.js [hunk 7]
```diff
@@ -6,10 +6,11 @@ Define a method "makeFriend" to add a new friend to her list.
 
 var person = {
     name: "Alice",
-    friends: ["John", "Nina"]
+    friends: ["John", "Nina"],
+    makeFriend: function (newFriends) {
+        friends = newFriends;
```
Comment (foaly-nr1) [inline week-4/D-methods/exercise-4.js]: Same as above. Can you spot the bug?

File: week-4/E-arrays-of-objects/exercise-2.js [hunk 8]
```diff
@@ -1,56 +1,68 @@
...
-
-var destinationNamesWithin500Kms = // Complete here
-
-var destinationNameReachableByFerry = // Complete here
-
-var destinationNamesMoreThan300KmsAwayByTrain = // Complete here (PRINT THE RESULT IN THE CONSOLE USING FOREACH)
-
-
+var destinationNamesWithin500Kms = travelDestinations
+    .filter(destination => destination.distanceKms <= 500)
```
Comment (foaly-nr1) [inline week-4/E-arrays-of-objects/exercise-2.js]: Another ES6 feature. Good work using an arrow function here.

File: week-4-practise/level-1/exercise-1.js (hunk 1 above)
Comment (clems27) [reply week-4-practise/level-1/exercise-1.js]: @foaly-nr1 thanks, I appreciate your comment

File: week-4-practise/level-2/exercise-1.js (hunk 2 above)
Comment (clems27) [reply week-4-practise/level-2/exercise-1.js]: Oh, thanks for pointing that out, I could do... return (this.volume = this.volume +100); Will make the correction

File: week-4/C-more-complex-objects/exercise-1.js (hunk 3 above)
Comment (clems27) [reply week-4/C-more-complex-objects/exercise-1.js]: @foaly-nr1 , Sure will do

File: week-4/C-more-complex-objects/exercise-1.js (hunk 4 above)
Comment (clems27) [reply week-4/C-more-complex-objects/exercise-1.js]: @foaly-nr1, maybe I should change the name to newOwnerAddress?

File: week-4/C-more-complex-objects/exercise-3.js (hunk 5 above)
Comment (clems27) [reply week-4/C-more-complex-objects/exercise-3.js]: When I tried calling the function with KinningParkHouse, that is an error .... cannot ready properties of undefined, I have used the plus (+), operator instead

File: week-4/D-methods/exercise-3.js (hunk 6 above)
Comment (clems27) [reply week-4/D-methods/exercise-3.js]: @foaly-nr1, thanks for point that out, I spotted the bug and I think I have fixed it, will be pushing the correction shortly

File: week-4/D-methods/exercise-4.js (hunk 7 above)
Comment (clems27) [reply week-4/D-methods/exercise-4.js]: Will make corrections

File: week-4/E-arrays-of-objects/exercise-2.js (hunk 8 above)
Comment (clems27) [reply week-4/E-arrays-of-objects/exercise-2.js]: Thanks

**Pull request closed without merging**